controllers/admin/AdminPsThemeCustoAdvanced.php

```diff
@@ -80,18 +83,100 @@ public function initContent()
+
+        if (!$bCreateChildTheme) {
+            die(false);
         }
 
-        $exporter = $kernel->getContainer()->get('prestashop.core.addon.theme.exporter');
-        $path = $exporter->export($this->context->shop->theme);
-        $aPath = array_reverse(explode('/', $path));
-        $sThemeZipPath = $this->module->ps_uri.'/themes/'.$aPath[0];
+        die(self::getChildTheme(_THEME_NAME_, _PS_ROOT_DIR_));
+    }
+
+    /**
+     * Prepare the child theme
+     *
+     * @param string $sParentThemeName
+     * @param string $sParentThemeDir
+     * @return bool
+    */
+    private function prepareChildTheme($sParentThemeName, $sParentThemeDir)
+    {
+        Tools::ZipExtract($this->childtheme_skeleton, $this->sandbox_path);
+
+        $aStringToReplace = array(
+            '{childtheme_parent}' => $sParentThemeName,
+            '{childtheme_name}' => 'child_'.$sParentThemeName,
+            '{childtheme_description}' => 'Child theme of '.$sParentThemeName.'\'s theme'
+        );
+
+        $sChildThemeConfigPath = $this->sandbox_path.'/'.$this->skeleton_name.'/config/theme.yml';
+
+        $sConfigFile = @file_get_contents($sChildThemeConfigPath);
+
+        foreach ($aStringToReplace as $sSearchElement => $sReplace) {
+            $sConfigFile = str_replace($sSearchElement, $sReplace, $sConfigFile);
+        }
+
+        $bPutContents = @file_put_contents($sChildThemeConfigPath, $sConfigFile);
+
+        if (!$bPutContents) {
+            return false;
+        }
+
+        $bCopyFile = @copy($sParentThemeDir.'/preview.png', $this->sandbox_path.'/'.$this->skeleton_name.'/preview.png');
+
+        if (!$bCopyFile) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Create the child theme
+     *
+     * @param string $sParentThemeName
+     * @param string $dest
+     * @return bool
+    */
+    private function createChildTheme($sParentThemeName)
+    {
+        $sChildThemeFolderName = 'child_'.$sParentThemeName;
+
+        $oZip = new ZipArchive;
+        $oZip->open($this->sandbox_path.'/'.$sChildThemeFolderName.'.zip', ZipArchive::CREATE);
+        $fileList = Finder::create()
+            ->files()
+            ->in($this->sandbox_path.'/'.$this->skeleton_name.'/');
+
+        foreach ($fileList as $file) {
+            $oZip->addFile($file->getRealpath(), $sChildThemeFolderName.'/'.$file->getRelativePathName());
+        }  
+        
+        return $oZip->close();
+    }
+
+    /**
+     * Move the ZIP archive into Theme's folder and unlink all the files in sandbox
+     *
+     * @param string $sParentThemeName
+     * @return string
+    */
+    private function getChildTheme($sParentThemeName, $sPrestashopRootDir)
+    {
+        $sChildThemeZipName = 'child_'.$sParentThemeName.'.zip';
+       
+        @copy($this->sandbox_path.'/'.$sChildThemeZipName, $sPrestashopRootDir.'/themes/'.$sChildThemeZipName);
+        @unlink($this->sandbox_path.'/'.$sChildThemeZipName);
```

Comment:
It seems you could directly move the file in the new folder with `rename()`, instead of copying / deleting

http://php.net/manual/fr/function.rename.php